Sources/SwiftInspectorVisitors/Tests/NestableTypeVisitorSpec.swift

```diff
@@ -81,6 +81,26 @@ final class NestableTypeVisitorSpec: QuickSpec {
               expect(classInfo?.name) == "SomeClass"
               expect(classInfo?.genericParameters.map { $0.name }) == ["T"]
             }
+
+            context("that is constrained with a where clause") {
```

Comment:
Would it be worth having a test somewhere to verify that we can find generic constraints when the type with the generic constraints is nested? Not sure it's valuable, so legitimately curious.

```swift
enum Parent {
  public class SomeClass<T>: SomeProtocol where T: CustomStringConvertible {
    public typealias SomeProtocolAssociatedType = A
  }
}
```

Reply:
I don't think we need this since the generic requirement is a property on the nestable type. We have tests showing that:
1. Deeply nested types are found and given the correct names
2. We can parse/find/surface generic requirements on nestable types

Between those two tests and the fact that our nested types are found via recursion, we can use induction to know that we can parse/find/surface generic requirements on deeply nested types.